provisioning/plugins/common/unsignedcorim_decoder.go

```diff
@@ -0,0 +1,91 @@
+package common
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/corim"
+	"github.com/veraison/endorsement"
+	"github.com/veraison/veraison/provisioning/decoder"
+)
+
+type IExtractor interface {
+	SwCompExtractor(comid.ReferenceValue) ([]*endorsement.SwComponent, error)
+	TaExtractor(comid.AttestVerifKey) (*endorsement.TrustAnchor, error)
+}
+
+func UnsignedCorimDecoder(data []byte, xtr IExtractor) (*decoder.EndorsementDecoderResponse, error) {
+	if len(data) == 0 {
+		return nil, errors.New("empty data")
+	}
+
+	var uc corim.UnsignedCorim
+
+	if err := uc.FromCBOR(data); err != nil {
+		return nil, fmt.Errorf("CBOR decoding failed: %w", err)
+	}
+
+	if err := uc.Valid(); err != nil {
+		return nil, fmt.Errorf("invalid unsigned corim: %w", err)
+	}
+
+	// TODO(tho) check profile
```

Comment:
Should we have a safe check mechanism to compare the outer profile with the inner profile and in case of mis-match bail out after reporting error?

Reply:
I am tempted to remove the TODO here

Reply:
Opened https://github.com/veraison/veraison/issues/112